### ISSUE TITLE
Add unit tests

### ISSUE THREAD

**Issue by ipfaze** [Scalingo/php-buildpack#364]
As discussed in PR https://github.com/Scalingo/php-buildpack/pull/341, adding unit tests would be very useful as we will be able to maintain and improve more easily our php-buildpack.

**Comment by ipfaze** [comment]
Issue https://github.com/Scalingo/php-buildpack/issues/236 is currently `🪦 Stale` waiting for this one to be tackled.

**Comment by Frzk** [comment]
First step consists in:

- [ ] adding a basic unit test using `shUnit2`
- [ ] making it runnable with `make test`
- [ ] including the process in a GitHub Actions Workflow
- [ ] making it run on both `scalingo-20` and `scalingo-22`

**Comment by Frzk** [comment]
Maybe we can close this one since we have a few tests now?
If not, could you please make it more specific (what exactly is expected?) ? Thanks!

**Comment by EtienneM** [comment]
I think we should close this issue when tests are automatically executed by the CI. Is it the case?

**Comment by Frzk** [comment]
The CI is triggered when a PR is opened: https://github.com/Scalingo/php-buildpack/actions
Also, first unit tests were introduced in https://github.com/Scalingo/php-buildpack/pull/369

**Comment by Frzk** [comment]
Currently the CI is triggered when the following events are happening:

- a `push` event occurs on the `main` branch: it doesn't really make any sense, since we are using the `master` branch.
- a `pull_request` event occurs on the repo: the workflow runs when a pull request is **opened** or **reopened** or **when the head branch of the pull request is updated** (a commit is made on the branch from which the PR is made) (full doc here: https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request)
- a `workflow_dispatch` event occurs, which means someone pushed the button to trigger it manually.

I guess we could either remove the `push` event or, if we want to keep it, make sure it's triggered when a push is done on the `master` branch.

WDYT?

**Comment by EtienneM** [comment]
Yes I think we need to update the `push` event so that it triggers for push on the `master` branch. Then we can close this issue
